File: src/main/java/client/inventory/operations/UpdateQuantityInventoryOperation.java
package client.inventory.operations;

import client.inventory.ItemInventoryType;
import client.inventory.ModifyInventoryOperationType;
import util.packet.PacketWriter;

public class UpdateQuantityInventoryOperation extends AbstractModifyInventoryOperation {

    private final short quantity;

    public UpdateQuantityInventoryOperation(ItemInventoryType inventory, short slot, short quantity) {
        super(inventory, slot);
        this.quantity = quantity;
    }

    @Override
    protected ModifyInventoryOperationType getType() {
        return ModifyInventoryOperationType.UPDATE_QUANTITY;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        pw.writeShort(quantity);
    }
}

File: src/main/java/client/inventory/operations/UpdateQuantityInventoryOperation.kt
package client.inventory.operations

import client.inventory.ItemInventoryType
import client.inventory.ModifyInventoryOperationType
import util.packet.PacketWriter

class UpdateQuantityInventoryOperation(inventory: ItemInventoryType, slot: Short, private val quantity: Short) : AbstractModifyInventoryOperation(inventory, slot) {

    override val type = ModifyInventoryOperationType.UPDATE_QUANTITY

    override fun encodeData(pw: PacketWriter) {
        pw.writeShort(quantity)
    }
}